l1-contracts/test/foundry/unit/concrete/Merkle/Merkle.t.sol

```diff
@@ -0,0 +1,119 @@
+        merkleTest = new MerkleTest();
+
+        elements.push(keccak256(hex"41"));
+        elements.push(keccak256(hex"42"));
+        elements.push(keccak256(hex"43"));
+        elements.push(keccak256(hex"44"));
+        elements.push(keccak256(hex"45"));
+        elements.push(keccak256(hex"46"));
+        elements.push(keccak256(hex"47"));
+        elements.push(keccak256(hex"48"));
+        elements.push(keccak256(hex"49"));
+        elements.push(keccak256(hex"4a"));
+        elements.push(keccak256(hex"4b"));
+        elements.push(keccak256(hex"4c"));
+        elements.push(keccak256(hex"4d"));
+        elements.push(keccak256(hex"4e"));
+        elements.push(keccak256(hex"4f"));
+        elements.push(keccak256(hex"50"));
+        elements.push(keccak256(hex"51"));
+        elements.push(keccak256(hex"52"));
+        elements.push(keccak256(hex"53"));
+        elements.push(keccak256(hex"54"));
+        elements.push(keccak256(hex"55"));
+        elements.push(keccak256(hex"56"));
+        elements.push(keccak256(hex"57"));
+        elements.push(keccak256(hex"58"));
+        elements.push(keccak256(hex"59"));
+        elements.push(keccak256(hex"5a"));
+        elements.push(keccak256(hex"61"));
+        elements.push(keccak256(hex"62"));
+        elements.push(keccak256(hex"63"));
+        elements.push(keccak256(hex"64"));
+        elements.push(keccak256(hex"65"));
+        elements.push(keccak256(hex"66"));
+        elements.push(keccak256(hex"67"));
+        elements.push(keccak256(hex"68"));
+        elements.push(keccak256(hex"69"));
+        elements.push(keccak256(hex"6a"));
+        elements.push(keccak256(hex"6b"));
+        elements.push(keccak256(hex"6c"));
+        elements.push(keccak256(hex"6d"));
+        elements.push(keccak256(hex"6e"));
+        elements.push(keccak256(hex"6f"));
+        elements.push(keccak256(hex"70"));
+        elements.push(keccak256(hex"71"));
+        elements.push(keccak256(hex"72"));
+        elements.push(keccak256(hex"73"));
+        elements.push(keccak256(hex"74"));
+        elements.push(keccak256(hex"75"));
+        elements.push(keccak256(hex"76"));
+        elements.push(keccak256(hex"77"));
+        elements.push(keccak256(hex"78"));
+        elements.push(keccak256(hex"79"));
+        elements.push(keccak256(hex"7a"));
+        elements.push(keccak256(hex"30"));
+        elements.push(keccak256(hex"31"));
+        elements.push(keccak256(hex"32"));
+        elements.push(keccak256(hex"33"));
+        elements.push(keccak256(hex"34"));
+        elements.push(keccak256(hex"35"));
+        elements.push(keccak256(hex"36"));
+        elements.push(keccak256(hex"37"));
+        elements.push(keccak256(hex"38"));
+        elements.push(keccak256(hex"39"));
+        elements.push(keccak256(hex"2b"));
+        elements.push(keccak256(hex"2f"));
+
+        root = merkleTree.getRoot(elements);
+    }
+
+    function testElements(uint256 i) public {
```

Comment:
Does it use boundary values? (i.e. the last and the first element)? I understand that it does multiple runs with the various values, but if it does not include the edge cases by default, it is better to include those manually

Reply:
i believe it does but will add explicit tests just to ensure